### PR TITLE
Homework 28.2

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp [hunk 1]
```diff
@@ -50,10 +51,6 @@
     </tbody>
     </tr>
 </table>
-<p>
-    <a href="${pageContext.request.contextPath}/" class="btn btn-secondary" role="button" aria-pressed="true">back to
-        main page</a>
-</p>
 
 </body>
 </html>
```
Comment (Rudiachenko) [inline src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp]: No new line at end of file

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp [hunk 2]
```diff
@@ -58,10 +59,5 @@
         <a href="${pageContext.request.contextPath}/products" class="btn btn-info" role="button" aria-pressed="true">back to all items page</a>
     </p>
 </table>
-<p>
-    <a href="${pageContext.request.contextPath}/" class="btn btn-secondary" role="button" aria-pressed="true">back to
-        main page</a>
-</p>
-
 </body>
 </html>
```
Comment (Rudiachenko) [inline src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: No new line at end of file

**Review by Rudiachenko — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
 @WebServlet("/shopping-cart/product/add")
 public class AddProductToShoppingCartController extends HttpServlet {
-    private static final Long USER_ID = 1L;
+    private static final String USER_ID = "user_id";
```
Comment (voloshynamaryna11) [inline src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java:17]: у меня у самой так же, но скорее всего скажут убрать это поле

**Review by voloshynamaryna11 — COMMENTED** [review]
done

File: src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp (hunk 1 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/admin/allProductsAdmin.jsp]: done

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp (hunk 2 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: done

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java:17]: Оки. посмотрим.

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java [hunk 4]
```diff
@@ -24,10 +24,9 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
-        String userId = req.getParameter("id");
-        Long id = Long.valueOf(userId);
-        Product product = productService.getById(id);
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        ShoppingCart shoppingCart = shoppingCartService.getByUserId(userId);
+        Product product = productService.getById(userId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java]: reconsider line 29 please

File: src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java [hunk 5]
```diff
@@ -24,10 +24,9 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
-        String userId = req.getParameter("id");
-        Long id = Long.valueOf(userId);
-        Product product = productService.getById(id);
-        ShoppingCart shoppingCart = shoppingCartService.getByUserId(USER_ID);
+        Long userId = (Long) req.getSession().getAttribute(USER_ID);
+        Product product = productService.getById(userId);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java]: same for line 28 here

File: src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java [hunk 6]
```diff
@@ -9,18 +9,15 @@
...
-    private static final String MESSAGE_ERROR = "Incorrect username or password";
     @Inject
     private UserService userService;
 
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
-                new AuthenticationException(MESSAGE_ERROR));
-
+        User userFromDB = userService.findByLogin(login).orElseThrow();
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java]:
```suggestion
        Optional<User> userFromDB = userService.findByLogin(login);
```

File: src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java [hunk 7]
```diff
@@ -9,18 +9,15 @@
...
         if (userFromDB.getPassword().equals(password)) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java]: let's add one more condition here

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp [hunk 8]
```diff
@@ -35,7 +36,7 @@
 
         <td>
             <p class="text-center">
-                <c:out value="${orders}"/>
+                <c:out value="${order.products}"/>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: are you sure you need this column? you already have `details` button

File: src/main/webapp/WEB-INF/views/index.jsp [hunk 9]
```diff
@@ -37,16 +37,19 @@
            href="#" role="button" aria-haspopup="true"
            aria-expanded="false">SHOW(ADMIN)</a>
         <div class="dropdown-menu">
-            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/products">USER</a>
-            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/orders">PRODUCT</a>
+            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/orders">USER</a>
+            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/products">PRODUCT</a>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/index.jsp]:
```suggestion
            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/orders">VIEW ORDERS</a>
            <a class="dropdown-item" href="${pageContext.request.contextPath}/admin/products">MANAGE PRODUCTS</a>
```

what do you think?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/cart/AddProductToShoppingCartController.java]: done. 
add  
String id = req.getParameter("id");
Long productId = Long.valueOf(id);

File: src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java (hunk 5 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/controllers/cart/DeleteProductsFromShoppingCartController.java]: done

File: src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java]: done

File: src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java (hunk 7 above)
Comment (vremere-a) [reply src/main/java/com/internet/shop/security/implementations/AuthenticationServiceImpl.java]: done

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp (hunk 8 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: agree

File: src/main/webapp/WEB-INF/views/index.jsp (hunk 9 above)
Comment (vremere-a) [reply src/main/webapp/WEB-INF/views/index.jsp]: you are right!
done

File: src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp [hunk 10]
```diff
@@ -34,11 +34,6 @@
             </p>
         </td>
 
-        <td>
-            <p class="text-center">
-                <c:out value="${order.products}"/>
-            </p>
-        </td>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/admin/allOrdersAdmin.jsp]: let's also remove `<th scope="col">Items</th>` this part above

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**